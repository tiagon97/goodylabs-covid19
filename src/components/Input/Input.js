import React from 'react';

const Input = ({ onChange }) => <input onChange={onChange} type="text" />;

export default Input;
