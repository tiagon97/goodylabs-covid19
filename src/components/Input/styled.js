import styled from 'styled-components';

import magnifierIcon from '../../assets/magnifier.svg';

export const InputWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin: 60px 0;
`;

export const StyledInput = styled.input`
  padding: 20px 80px 20px 40px;
  font-size: 1.6rem;
  background-color: #ddd;
  border: none;
  border-radius: 50px;
  background-image: url(${magnifierIcon});
  background-size: 15px;
  background-position: 15px 50%;
  background-repeat: no-repeat;
  outline: none;

  &::placeholder {
    text-transform: uppercase;
    font-size: 1.2rem;
    letter-spacing: 1px;
    color: #bbb;
  }
`;
