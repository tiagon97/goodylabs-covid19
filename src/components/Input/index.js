import React from 'react';
import PropTypes from 'prop-types';

import { InputWrapper, StyledInput } from './styled';

const Input = ({ onChange, value, name }) => (
  <InputWrapper>
    <StyledInput value={value} placeholder={name} onChange={onChange} name={name} id={name} type="text" />
  </InputWrapper>
);

export default Input;

Input.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};
