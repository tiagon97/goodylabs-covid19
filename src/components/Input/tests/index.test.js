/**
 *
 * Tests for Input
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import Input from '../index';

const defaultProps = {
  onChange: jest.fn(),
  value: 'Value',
  name: 'Name',
};

describe('<Input />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Input {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Input {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
