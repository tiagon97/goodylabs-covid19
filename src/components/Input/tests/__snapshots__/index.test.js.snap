// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render and match the snapshot 1`] = `
<div
  class="sc-AxjAm WWPws"
>
  <input
    class="sc-AxirZ jUgtCW"
    id="Name"
    name="Name"
    placeholder="Name"
    type="text"
    value="Value"
  />
</div>
`;
